crates/cxx-qt-gen/src/writer/cpp/mod.rs

```diff
@@ -300,10 +300,10 @@ mod tests {
         class MyObject : public QStringListModel
         {
           Q_OBJECT
+        public:
           Q_PROPERTY(int count READ count WRITE setCount NOTIFY countChanged)
```

Comment:
why are the metaobjects now public ? Normally these are above aren't they ? 

eg https://doc.qt.io/qt-6/properties.html#a-simple-example shows the `Q_OBJECT`, `Q_PROPERTY` above the `public:` and the enum and `Q_ENUM` below ?

Maybe just adding to `methods` is the right way to do the enum? even though it isn't a "method" but it's something in the public space ? (maybe we need a better name there)

Reply:
The reason for this is that we really want the enum class itself (or rather the `using` directive for it) to be public in the class (as you correctly guessed).

I've considered putting the `Q_ENUM` into another part of the generation, but honestly that seemed like a bit of a hack.
As a `Q_ENUM` is as much a meta-object as `Q_PROPERTY` I think it should go into the same category.

So the simplest solution is to move all metaobjects into the public category.
This doesn't change any behavior, as Q_PROPERTY is just defined as nothing in C++.
And semantically one could argue that it's correct anyhow, as Q_PROPERTY values are indeed publicly accessible.
It does go a bit against convention, though there don't seem to be any strong guidelines or reasoning that support this convention.

Reply:
Note that `Q_OBJECT` and `Q_GADGET` expand to something that has as `private:` at the end, we would need to assume here that `Q_PROPERTY` would not also have any `private:` or `protected:` in it's expansion (which it doesn't currently).

Reply:
As that would be breaking change, this can't happen until Qt7, so let's not worry about this for now.